jdtls.ext/com.microsoft.jdtls.ext.core/src/com/microsoft/jdtls/ext/core/PackageCommand.java

```diff
@@ -577,26 +577,17 @@ private static Object[] findJarDirectoryChildren(JarEntryDirectory directory, St
 
     public static IProject getProject(String projectUri) {
         IWorkspaceRoot root = ResourcesPlugin.getWorkspace().getRoot();
-        IContainer[] containers = root.findContainersForLocationURI(JDTUtils.toURI(projectUri));
+        URI uri = JDTUtils.toURI(projectUri);
+        IContainer[] containers = root.findContainersForLocationURI(uri);
 
-        if (containers.length == 0) {
-            return null;
-        }
-
-        // For multi-module scenario, findContainersForLocationURI API may return a container array,
-        // put the result from the nearest project in front.
-        Arrays.sort(containers, (Comparator<IContainer>) (IContainer a, IContainer b) -> {
-            return a.getFullPath().toPortableString().length() - b.getFullPath().toPortableString().length();
-        });
-
-        for (IContainer container : containers) {
-            IProject project = container.getProject();
-            if (!project.exists()) {
-                return null;
-            }
-            return project;
+        Optional<IContainer> maybeProject = Arrays.stream(containers).filter(container -> container instanceof IProject).findFirst();
+        if (maybeProject.isPresent()) {
+            return (IProject) maybeProject.get();
+        } else {
+            // This must be a folder that has been linked to project outside the workspace
+            // for example an invisible project or a project created by a third party extension. 
+            return containers[0].getProject();
```

Comment:
nit: Add a length check for the `containers`.

Reply:
@jdneo, I added some error handling. I also added back the sorting, because nothing would stop a third party extension from creating invisible projects anywhere in the hierarchy (not just at the root) and then we need to decide which folder to use.

Reply:
Not quite sure if the sorting logic is correct. Because the project name could be very long or very short which makes the length of the `fullPath` unpredictable.

Anyway, I decide to merge it first, since in the cases we know for now, it works well. We can revisit this later if cases we found fails.

Maybe, the project's `fullPath` should be used here instead of the uri?

Reply:
@jdneo, you may be right. I was thinking that a third party extension could create the following setup:
```
root
  project
    folder
```
`root` and `project` are projects and `folder` is an invisible project. `root.findContainersForLocationURI(uri)` will probably find:
- F/root/project/folder
- F/project/folder
But I guess it doesn't matter which one we pick, they should all link to the same project.

I just removed the sort.